### PR TITLE
chore: E2E tests for the Drill to detail modal

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
This PR adds E2E tests for the drill to detail modal originally implemented in this PR https://github.com/apache/superset/pull/20728

Related RTL tests: https://github.com/apache/superset/pull/21148

It also adds a new ECharts Dashboard sample with some sample ECharts (as shown in the screenshot). The goal of this Dashboard is to grow and eventually show as many ECharts samples as possible. I'll continue enhancing this sample dashboard in separate PRs.

ECharts Dashboard

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A.

### TESTING INSTRUCTIONS
All the tests should succeed

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/cypress-base/cypress/integration/dashboard/drilltodetail.test.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+describe('Drill to detail modal', () => {
+  beforeEach(() => {
+    cy.login();
+    cy.visit(ECHARTS_DASHBOARD);
+    ECHARTS_CHARTS.forEach(waitForChartLoad);
+  });
+
+  it('opens the modal from the context menu', () => {
+    cy.get(
```
Comment (michael-s-molina) [inline superset-frontend/cypress-base/cypress/integration/dashboard/drilltodetail.test.ts:37]: Can we extract the code that opens the modal to a function? Specifically this part:

```
cy.get(
  "[data-test-viz-type='big_number_total'] [aria-label='More Options']",
).click();
cy.get('.ant-dropdown')
  .not('.ant-dropdown-hidden')
  .find("[role='menu'] [role='menuitem']")
  .eq(5)
  .should('contain', 'Drill to detail')
  .click();
```

Comment (jinghua-qa) [reply superset-frontend/cypress-base/cypress/integration/dashboard/drilltodetail.test.ts:37]: +1

Comment (geido) [reply superset-frontend/cypress-base/cypress/integration/dashboard/drilltodetail.test.ts]: Sure. Updated

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Review by jinghua-qa — APPROVED** [review]
LGTM

**Pull request merged**